addons/account/models/account_move.py

```diff
@@ -1370,6 +1368,9 @@ def _compute_bank_partner_id(self):
             else:
                 move.bank_partner_id = move.company_id.partner_id
 
+            bank_ids = move.bank_partner_id.bank_ids.filtered(lambda bank: bank.company_id in (False, move.company_id))
+            move.partner_bank_id = bank_ids and bank_ids[0]
```

Comment:
Also, if you go this way, `partner_bank_id` should have its own compute method and being plugged on a compute from another field. (even when it's tightly related)